### PR TITLE
Add a machine translation option when syncing pages

### PR DESCRIPTION
**Pull request by waldo90**
This PR adds an option for machine translation to be performed when synicing page translations as per issue:
[#786](https://github.com/wagtail/wagtail-localize/issues/786#issue-2169754390)

It adds the option alongside the existing 'publish immediately' option as I found that these can be made to work togther in all combinations:
 - sync only
 - sync and publish without machine translation
 - sync and publish with machine tranlation
 - sync with machine translation (unpublished but ready for review)

This is slightly as odds with what was discussed on the ticket. However, I'm hopeful that the extra flexibility will be welcome.

### CODE REVIEW

File: wagtail_localize/views/update_translations.py [hunk 1]
```diff
@@ -17,16 +17,35 @@
...
+if get_machine_translator():
+    PUBLISH_TRANSLATIONS_HELP = (
+        "Select this to publish immediately. Changes will be published "
+        "in the original language unless you also select use machine translation."
+    )
+else:
+    PUBLISH_TRANSLATIONS_HELP = (
+        "Select this to publish immediately. Changes will be published "
+        "in the original language until you translate and publish them."
+    )
```
Comment (zerolab) [inline wagtail_localize/views/update_translations.py:35]: I think these need tweaking as there is a discrepancy in style between this and the use machine translation help.

I suggest something along the lines of:

> Apply the updates and publish immediately. The changes will use the original language until translated unless you also select "Use machine translation".

and 

> Apply the updates and publish immediately. The changes will use the original language until translated.

File: wagtail_localize/views/edit_translation.py [hunk 2]
```diff
@@ -1426,12 +1425,19 @@ def machine_translate(request, translation_id):
...
                             "field_error": "",
                         },
                     )
+        return True
+    return False
 
+
+@require_POST
+def machine_translate(request, translation_id):
+    if apply_machine_translation(translation_id, request.user):
```
Comment (zerolab) [inline wagtail_localize/views/edit_translation.py:1439]: `apply_machine_translation` is doing too many things and can have unintended consequences on the sync translation form.

I would rather it purely takes the translation id, user, and the machine translator instance and it *applies* the machine translation.

Move the additional checks in the view (and ditto, as additional checks on the sync translation form init and validation)

**Review by zerolab — CHANGES_REQUESTED** [review]
This is a great start @waldo90 

Left a few notes and suggestions for improvement

File: wagtail_localize/views/update_translations.py (hunk 1 above)
Comment (waldo90) [reply wagtail_localize/views/update_translations.py:35]: Thank you, I've used your copy verbatim.

File: wagtail_localize/views/update_translations.py [hunk 3]
```diff
@@ -17,16 +17,35 @@
...
 class UpdateTranslationsForm(forms.Form):
     publish_translations = forms.BooleanField(
         label=gettext_lazy("Publish immediately"),
-        help_text=gettext_lazy(
-            "This will apply the updates and publish immediately, before any new translations happen."
-        ),
+        help_text=gettext_lazy(PUBLISH_TRANSLATIONS_HELP),
+        required=False,
+    )
+    use_machine_translation = forms.BooleanField(
```
[the remark this answers is not in the document]
Comment (waldo90) [reply wagtail_localize/views/update_translations.py:47]: The checkbox rendering is bypassed in the template `update_translations.html` around line 53 which checks if the machine translator is available. It's a different approach to what you are thinking but what do you think? Is that ok?

[the remark this answers is not in the document]
Comment (waldo90) [reply wagtail_localize/views/update_translations.py:49]: Added tests foor the two form states

**Comment by zerolab** [comment]
Thanks @waldo90. Will finish the work in #842 (mainly rebase and a bit of tidy up)

**Pull request closed without merging**